src/v/kafka/server/tests/fetch_plan_bench.cc

```diff
@@ -0,0 +1,213 @@
+#include "model/namespace.h"
+#include "random/generators.h"
+#include "redpanda/tests/fixture.h"
+#include "test_utils/fixture.h"
+
+#include <seastar/core/sstring.hh>
+#include <seastar/testing/perf_tests.hh>
+#include <seastar/testing/thread_test_case.hh>
+
+#include <boost/range/iterator_range_core.hpp>
+#include <boost/test/tools/interface.hpp>
+#include <boost/test/tools/old/interface.hpp>
+#include <boost/test/unit_test_log.hpp>
+#include <fmt/ostream.h>
+
+#include <tuple>
+
+static ss::logger fpt_logger("fpt_test");
+
+using namespace std::chrono_literals; // NOLINT
+struct fixture {
+    static kafka::fetch_session_partition make_fetch_partition(
+      model::topic topic, model::partition_id p_id, model::offset offset) {
+        return kafka::fetch_session_partition{
+          .topic_partition = {std::move(topic), p_id},
+          .max_bytes = 1_MiB,
+          .fetch_offset = offset,
+          .high_watermark = offset};
+    }
+
+    static kafka::fetch_request::topic
+    make_fetch_request_topic(model::topic tp, int partitions_count) {
+        kafka::fetch_request::topic fetch_topic{
+          .name = std::move(tp),
+          .fetch_partitions = {},
+        };
+
+        for (int i = 0; i < partitions_count; ++i) {
+            fetch_topic.fetch_partitions.push_back(
+              kafka::fetch_request::partition{
+                .partition_index = model::partition_id(i),
+                .fetch_offset = model::offset(i * 10),
+                .max_bytes = 100_KiB,
+              });
+        }
+        return fetch_topic;
+    }
+};
+
+struct fetch_plan_fixture : redpanda_thread_fixture {
+    static constexpr size_t topic_name_length = 30;
+    static constexpr size_t total_partition_count = 800;
+    static constexpr size_t session_partition_count = 100;
+
+    model::topic t;
+
+    fetch_plan_fixture() {
+        BOOST_TEST_CHECKPOINT("before leadership");
+
+        wait_for_controller_leadership().get0();
+
+        BOOST_TEST_CHECKPOINT("HERE");
+
+        t = model::topic(
+          random_generators::gen_alphanum_string(topic_name_length));
+        auto tp = model::topic_partition(t, model::partition_id(0));
+        add_topic(
+          model::topic_namespace_view(model::kafka_namespace, t),
+          total_partition_count)
+          .get();
+
+        BOOST_TEST_CHECKPOINT("HERE");
+    }
+};
+
+PERF_TEST_F(fetch_plan_fixture, test_fetch_plan) {
+    // make the fetch topic
+    kafka::fetch_topic ft;
+    ft.name = t;
+
+    // add the partitions to the fetch request
+    for (int pid = 0; pid < session_partition_count; pid++) {
+        kafka::fetch_partition fp;
+        fp.partition_index = model::partition_id(pid);
+        fp.fetch_offset = model::offset(0);
+        fp.current_leader_epoch = kafka::leader_epoch(-1);
+        fp.log_start_offset = model::offset(-1);
+        fp.max_bytes = 1048576;
+        ft.fetch_partitions.push_back(std::move(fp));
+    }
+
+    BOOST_TEST_CHECKPOINT("HERE");
```

Comment:
interesting, what's this? does it print out some context in the log to help orient oneself when examining the output?

Reply:
Yeah I had crashes during fixture startup or in the test (this was originally written as a fixture test), but you basically get zero context when that happens because an exception is thrown and then there's either a crash during unwinding the stack) or the exception makes it all the way out but doesn't contain actionable info.

When it happens boost reports the last "checkpoint" which are things like BOOST_CHECK macros, the start of the test, etc. This just adds a new checkpoint to help you isolate the crash point. 

Kind of hacky. I can remove them if you want.

They all say "here" since the line is included so there's little point in coming up with a clever name.

Reply:
nah no need to remove. was just wondering